src/redux/books/books.js

```diff
@@ -1,17 +1,30 @@
-// Actions
-const actionName = 'project-name/folder/name';
+const initialState = [];
+// Actions Types
+const ADD_BOOK = 'bookStore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
 
 // Reducer
-export default function reducer(state = {}, action = {}) {
+const reducer = (state = initialState, action) => {
   switch (action.type) {
-    // do reducer stuff
-    default: return state;
+    case ADD_BOOK:
+      return [...state, action.payload];
+    case REMOVE_BOOK:
+      return state.filter(book => book.id !== action.payload);
+    default:
+      return state;
   }
-}
+};
+export default reducer;
+
 // Action Creators
-export function loadWidgets() {
-  return { type: actionName };
-}
+export const addBook = payload => ({
+  type: ADD_BOOK,
+  payload,
+});
+export const removeBook = payload => ({
+  type: REMOVE_BOOK,
+  payload,
+});
```

Comment:
- [ ] Let us create the actions before anything and call the actions created in the reducers, refer to the below example.

const ADD_BOOK = 'bookStore/books/ADD_BOOK';
const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';

const initialState = [];

export const addBook = (payload) => ({
  type: ADD_BOOK,
  payload,
});

export const removeBook = (payload) => ({
  type: REMOVE_BOOK,
  payload,
});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_BOOK:
      return [...state, action.payload];
    case REMOVE_BOOK:
      return state.filter((book) => book.id !== action.payload.id);

    default:
      return state;
  }
};

export default reducer;